client/features/data-conversion.js

```diff
@@ -5,31 +5,36 @@ export const convertEventPayloadsWithRemoteEncoder = async (events, endpoint) =>
   const requests = [];
 
   events.forEach(event => {
-    let payloads = [];
+    let payloadsWrapper;
 
     if (event.details.input) {
-      payloads = event.details.input.payloads;
+      payloadsWrapper = event.details.input;
     } else if (event.details.result) {
-      payloads = event.details.result.payloads;
+      payloadsWrapper = event.details.result;
     }
 
-    payloads.forEach((payload, i) => {
-      requests.push(
-        fetch(`${endpoint}/decode`, { method: 'POST', headers: headers, body: JSON.stringify(payload) })
+    if (!payloadsWrapper) {
+      return;
+    }
+
+    requests.push(
+      fetch(`${endpoint}/decode`, { method: 'POST', headers: headers, body: JSON.stringify(payloadsWrapper) })
```

Comment:
Just a thought we should make sure wherever this is happening has http2.1 otherwise we'll only be able to decode 8 things at a time which would severely block rendering.

Reply:
The server aspect will be up to the users, it's not something we control. We can document that it should support http2  though.